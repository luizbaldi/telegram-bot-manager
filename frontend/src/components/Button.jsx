import React from 'react'
import styled from 'styled-components'

const Button = ({ children, onClick, block }) => (
  <Container onClick={onClick} block={block}>
    {children}
  </Container>
)

/* Styled components */
const Container = styled.div`
  background-color: papayawhip;
  padding: 12px;
  border-radius: 2px;
  display: ${({ block }) => block ? 'block' : 'inline-block'}
  text-align: center;

  &:hover {
    background-color: palegoldenrod;
    cursor: pointer;
  }
`

export default Button
