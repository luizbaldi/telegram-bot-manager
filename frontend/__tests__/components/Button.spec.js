import React from 'react'
import { render } from 'react-testing-library'
import Button from '../../src/components/Button'

test('<Button />', () => {
  const { container } = render((
    <Button>
      Potatoe
    </Button>
  ))

  expect(container).toMatchSnapshot()
})
