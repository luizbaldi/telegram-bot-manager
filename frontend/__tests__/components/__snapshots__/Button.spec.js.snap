// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> 1`] = `
<div>
  <div
    class="sc-bdVaJa cKRTaz"
  >
    Potatoe
  </div>
</div>
`;
